libyul/AsmAnalysis.cpp

```diff
@@ -485,18 +487,38 @@ bool AsmAnalyzer::operator()(ForLoop const& _for)
 	if (!expectExpression(*_for.condition))
 		success = false;
 	m_stackHeight--;
+
+	m_info.stackHeightInfo[&_for] = m_stackHeight - scope(&_for.pre).numberOfVariables();
+
+	// backup outer for-loop & create new state
+	auto oldForLoop = exchange(m_currentForLoop, &_for);
```

Comment:
```suggestion
	auto oldForLoop = m_currentForLoop;
	m_currentForLoop = &_for;
```

Reply:
Also: "outer" instead of "old"?

Reply:
Or "original" (in line with "originalScope")?

Reply:
sounds better. thanks.